### ISSUE TITLE
--branch option for vendor pattern is broken [upstream]

### ISSUE THREAD

**Issue by ches** [websterclay/knife-github-cookbooks#7]
_This is not really a `knife-github-cookbooks` bug -- it's [broken in knife core](http://tickets.opscode.com/browse/CHEF-2105) and `knife-github-cookbooks` suffers for it. I'm shamelessly filing it here in hopes that it increases visibility to others who are bothered by it and that more votes on the Jira ticket or comments on the pull request might get Opscode's attention, and additionally to make anyone who may stumble across it aware that it is not @websterclay's bug..._

To summarize, the issue is that knife's [vendor branch pattern](https://github.com/websterclay/knife-github-cookbooks/issues/6) magic automatically checks out master and merges into it when you install/update a cookbook. I want to test and add configuration and customizations after installing a cookbook, and merge when it's ready for production, so this is highly annoying. A `--branch` option was added to Knife to address this, but it was broken in the push to Chef 0.10.0.

**Comment by ches** [comment]
Doh, sorry, I just saw that this was mentioned and closed in #6 since I thought of filing this here awhile ago, and that you opened a new Jira ticket for it. Maybe we'll have some luck in getting it merged with [CHEF-2105](http://tickets.opscode.com/browse/CHEF-2105)...

**Comment by ches** [comment]
This has been fixed upstream and released in Chef 0.10.4. Woot :-)
